internal/controller/apiextensions/claim/configurator.go

```diff
@@ -183,82 +179,94 @@ func filter(in map[string]any, keys ...string) map[string]any {
 	return out
 }
 
-// APIClaimConfigurator configures the supplied claims with fields
-// from the composite. This includes late-initializing spec values
-// and updating status fields in claim.
-type APIClaimConfigurator struct {
-	client client.Client
-}
+func keep(in map[string]any, keys ...string) map[string]any {
+	filter := map[string]bool{}
+	for _, k := range keys {
+		filter[k] = true
+	}
 
-// NewAPIClaimConfigurator returns a APIClaimConfigurator.
-func NewAPIClaimConfigurator(client client.Client) *APIClaimConfigurator {
-	return &APIClaimConfigurator{client: client}
+	out := map[string]any{}
+	for k, v := range in {
+		if filter[k] {
+			out[k] = v
+		}
+	}
+	return out
 }
 
-// Configure the supplied claims with fields from the composite.
+// configure the supplied claim patch with fields from the composite.
 // This includes late-initializing spec values and updating status fields in claim.
-func (c *APIClaimConfigurator) Configure(ctx context.Context, cm resource.CompositeClaim, cp resource.Composite) error {
-	ucm, ok := cm.(*claim.Unstructured)
-	if !ok {
-		return nil
+func configureClaim(_ context.Context, cm *claim.Unstructured, cmPatch *claim.Unstructured, cp *composite.Unstructured, cpPatch *composite.Unstructured) error { //nolint:gocyclo // Only slightly over (10)
```

Comment:
Sure, we can rename `cm` to `cmObserved`, but `cmDesired` is not the right name, because we only produce here a patch, not the complete desired resource.

Reply:
I am not sure I understand the difference between a patch vs complete desired resource considering we use `composite.Unstructured` as type 🤔